gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/cloud/storage/contrib/nio/package-info.java

```diff
@@ -0,0 +1,98 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Java 7 nio FileSystem client library for Google Cloud Storage.
+ *
+ * <p>This client library allows you to easily interact with Google Cloud Storage, using Java's
+ * standard file system API, introduced in Java 7.
+ *
+ * <h3>How It Works</h3>
+ *
+ * The simplest way to get started is with {@code Paths} and {@code Files}:<pre>   {@code
+ *
+ *   Path path = Paths.get(URI.create("gs://bucket/lolcat.csv"));
+ *   List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);}</pre>
+ *
+ * <p>If you want to configure the bucket per-environment, it might make more sense to use the
+ * {@code FileSystem} API:
+ * <pre>
+ *   class Foo {
+ *     static String bucket = System.getProperty(...);
+ *     static FileSystem fs = FileSystems.getFileSystem(URI.create("gs://" + bucket));
+ *     void bar() {
+ *       byte[] data = "hello world".getBytes(StandardCharsets.UTF_8);
+ *       Path path = fs.getPath("/object");
+ *       Files.write(path, data);
+ *       data = Files.readBytes(path);
+ *     }
+ *     void doodle() {
+ *       Path path = fs.getPath("/path/to/doodle.csv");
+ *       List&lt;String&gt; lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+ *     }
+ *   }</pre>
+ *
+ * <p>You can also use InputStream and OutputStream for streaming:
+ * <pre>
+ *   Path path = Paths.get(URI.create("gs://bucket/lolcat.csv"));
+ *   try (InputStream input = Files.openInputStream(path)) {
+ *     // ...
+ *   }</pre>
+ *
+ * <p>You can set various attributes using
+ * {@link com.google.cloud.storage.contrib.nio.CloudStorageOptions CloudStorageOptions} static
+ * helpers:
+ * <pre>
+ *   Files.write(csvPath, csvLines, StandardCharsets.UTF_8,
+ *       withMimeType(MediaType.CSV_UTF8),
+ *       withoutCaching());</pre>
+ *
+ * <p><b>NOTE:</b> Cloud Storage uses a flat namespace and therefore doesn't support real
+ * directories. So this library supports what's known as "pseudo-directories". Any path that
+ * includes a trailing slash, will be considered a directory. It will always be assumed to exist,
+ * without performing any I/O. This allows you to do path manipulation in the same manner as you
+ * would with the normal UNIX file system implementation. You can disable this feature with
+ * {@link com.google.cloud.storage.contrib.nio.CloudStorageConfiguration#usePseudoDirectories()}.
+ *
+ * <h3>Unit Testing</h3>
+ *
+ * <p>Here's a simple unit test:<pre>
+ *
+ *   class MyTest {
+ *     {@literal @}Rule
+ *     public final AppEngineRule appEngine = AppEngineRule.builder().build();
```

Comment:
As for removing this section, I guess we'll have to since `FakeStorageRpc` and `LocalStorageHelper` are now restricted to the package. Let's put it back once we expose those again since being able to test against a fake filesystem is a useful feature.